Add etc/ssdlc_compliance_report.md (CXX-3013, CXX-3024, and CXX-3039) (#1149)
mongodb/mongo-cxx-driver: etc/ssdlc_compliance_report.md

# Mongo C++ Driver SSDLC Compliance Report

## Release Creator

- See [C/CXX Release Info](https://docs.google.com/spreadsheets/d/1yHfGmDnbA5-Qt8FX4tKWC5xk9AhzYZx1SKF4AD36ecY/edit?usp=sharing).

## Process Document

> - See [go/how-we-develop-software-doc](http://go/how-we-develop-software-doc).

Reviewer: Suggest leaving as a placeholder.

```suggestion
Unavailable. <!-- TODO: Replace with link to document referenced in CXX-3007 once available. -->
```

I expect the `go` link is inaccessible to external users. [MongoDB Software Security Development Lifecycle Policy](https://docs.google.com/document/d/1u0m4Kj2Ny30zU74KoEFCN4L6D_FbEYCaJ3CQdCYXTMc/edit#bookmark=id.ic9o2y62f82x) notes:
> For products with publicly available source code, our practice of how we develop software is expected to be available in a public facing document.

Author: Expectation was that the GoLink will be updated to the public-facing document, but forgot that GoLinks themselves are internal-only. 🤦 Will remove reference as suggested.

## Tool used to track third party vulnerabilities

> - See [Silk](https://us1.app.silk.security/inventory/asset-group/mongodb____DedupedAssetGroup____60640b8853771efe3af5f78ea37af5d1cdd190df).

Author: Done.

- See [C++ Driver - SSDLC Reports](https://drive.google.com/drive/folders/1q9RI55trFzHlh8McALSIAbT6ugyn8zlO) for release-specific reports.

## Third-Party Dependency Information

- See [etc/augmented.sbom.json](https://github.com/mongodb/mongo-cxx-driver/blob/master/etc/augmented.sbom.json) within the release tarball.
- See [etc/third_party_vulnerabilities.md](https://github.com/mongodb/mongo-cxx-driver/blob/master/etc/third_party_vulnerabilities.md) within the release tarball.
- See [C++ Driver - SSDLC Reports](https://drive.google.com/drive/folders/1q9RI55trFzHlh8McALSIAbT6ugyn8zlO) for release-specific reports.

## Static Analysis Findings

- See [C++ Driver - SSDLC Reports](https://drive.google.com/drive/folders/1q9RI55trFzHlh8McALSIAbT6ugyn8zlO) for release-specific reports.

## Security Testing Report

> - See [Driver Security Testing Summary](https://docs.google.com/document/d/1y2K_RY4GZVXpQvv4JH_35mSzFRTawNJ3mibpvSBU8H0/edit?usp=sharing). Available upon request for approved partners (contact Rishabh Bisht \<rishabh.bisht@mongodb.com\>).

Reviewer: I expect the contact information was motivated [by this conversation](https://mongodb.slack.com/archives/C05N744KWQ4/p1717092734498109). However, I am mildly concerned about putting a direct e-mail (may be a target for spam). I expect customers have existing means to contact support. I suggest using wording matching [minimal.md](https://gist.github.com/rtimmons/8253d1dfcc5f233bdb2d9c73b8e5c129#file-minimal-md):

```suggestion
- See [Driver Security Testing Summary](https://docs.google.com/document/d/1y2K_RY4GZVXpQvv4JH_35mSzFRTawNJ3mibpvSBU8H0/edit?usp=sharing) (internal). Available as needed from the MongoDB C++ Driver team.
```

Author: Done.

> ## Security Assessment Report
> 
> - Available upon request for approved partners (contact Rishabh Bisht \<rishabh.bisht@mongodb.com\>).

Reviewer: ```suggestion
```

Suggest removing. The "Security Assessment By an External Team" is a non-goal in [Scope: SSDLC Policy Conformance for Client Libs](https://docs.google.com/document/d/1mXxDdY38nGVKWJhIbH83X6mcHsxllAiunH2hg-NdDsY/) so I expect there is no report to share.

Author: Added an explicit note stating this report is not applicable to the CXX Driver. LMK if you would prefer the section be removed entirely instead.

## Signature Information

- The source tarball for each release is accompanied by a detached GPG digital signature which may be verified against the `cpp-driver` public key available at https://pgp.mongodb.com/.

## Known Vulnerabilities

> - Any vulnerabilities that may be shown in the links referenced above have been reviewed and accepted by the appropriate approvers. For detailed information, [C++ Driver - SSDLC Reports](https://drive.google.com/drive/folders/1q9RI55trFzHlh8McALSIAbT6ugyn8zlO) for release-specific reports.

Author: Done.
